tests/pipeline/test_statistical.py

```diff
@@ -802,3 +879,139 @@ def test_factor_regression_method_two_factors(self, regression_length):
+        self.assertTrue((np.abs(result - true_betas) < 0.01).all())
+
+    @parameter_space(seed=[1, 2, 3], __fail_fast=True)
+    def test_nan_handling_matches_empyrical(self, seed):
+        rand = np.random.RandomState(seed)
+
+        true_betas = np.array([-0.5, 0.0, 0.5, 1.0, 1.5])
+        independent = as_column(np.linspace(-5., 5., 30))
+        noise = as_column(rand.uniform(-.1, .1, 30))
+        dependents = 1.0 + true_betas * independent + noise
+
+        # Fill 20% of the input arrays with nans randomly.
+        dependents[rand.uniform(0, 1, dependents.shape) < 0.2] = nan
+        independent[rand.uniform(0, 1, independent.shape) < 0.2] = nan
+
+        # Sanity check that we actually inserted some nans.
+        self.assertTrue(np.count_nonzero(np.isnan(dependents)) > 0)
+        self.assertTrue(np.count_nonzero(np.isnan(independent)) > 0)
+
+        result = self.compare_with_empyrical(dependents, independent)
+
+        # compare_with_empyrical uses requred_observations=0, so we shouldn't
+        # have any nans in the output even though we had some in the input.
+        self.assertTrue(not np.isnan(result).any())
+
+        # We should still have reasonable beta estimates.
+        self.assertTrue((np.abs(result - true_betas) < 0.01).all())
+
+    @parameter_space(nan_offset=[-1, 0, 1])
+    def test_produce_nans_when_too_much_missing_data(self, nan_offset):
+        rand = np.random.RandomState(42)
+
+        true_betas = np.array([-0.5, 0.0, 0.5, 1.0, 1.5])
+        independent = as_column(np.linspace(-5., 5., 30))
+        noise = as_column(rand.uniform(-.1, .1, 30))
+        dependents = 1.0 + true_betas * independent + noise
+
+        # Write nans in a triangular pattern into the middle of the dependent
+        # array.
+        nan_grid = np.array([[1, 0, 0, 0, 0],
+                             [1, 1, 0, 0, 0],
+                             [1, 1, 1, 0, 0],
+                             [1, 1, 1, 1, 0],
+                             [1, 1, 1, 1, 1]], dtype=bool)
+        num_nans = nan_grid.sum(axis=0)
+        # Move the grid around in the parameterized tests. The positions
+        # shouldn't matter.
+        dependents[10 + nan_offset:15 + nan_offset][nan_grid] = np.nan
+
+        for allowed_missing in range(7):
+            results = vectorized_beta(dependents, independent, allowed_missing)
+            for i, expected in enumerate(true_betas):
+                result = results[i]
+                expect_nan = num_nans[i] > allowed_missing
+                true_beta = true_betas[i]
+                if expect_nan:
+                    self.assertTrue(np.isnan(result))
+                else:
+                    self.assertTrue(np.abs(result - true_beta) < 0.01)
+
+    def test_allowed_missing_doesnt_double_count(self):
+        # Test that allowed_missing only counts a row as missing one
+        # observation if it's missing in both the dependent and independent
```

Comment:
should "both" be "either"? Dates for which the dependent or independent (or both) data is nan cannot be included in the beta calculation.

Reply:
I think this wording is correct; the thing that this test is checking is that if a row has a NaN in both the dependent and independent, then we only count that row as adding 1 toward the number of required missing values.